api/src/main/java/com/cloud/server/ResourceTag.java

```diff
@@ -91,6 +96,53 @@ public boolean resourceMetadataSupport() {
         public boolean resourceIconSupport() {
             return resourceIconSupport;
         }
+
+        static {
+            resourceObjectTypeMap.put("uservm", UserVm);
```

Comment:
@ravening is there any automatic way to have lowered-case object-type name map to the ResourceObjectType? Otherwise, this creates issue of manually maintaining this hashmap.

Reply:
@rohityadavcloud didnt get your point. can you explain more?

Reply:
For every ResourceObjectType, you're manually entering a map in resourceObjectTypeMap. Why not use reflections or other ways to generate this map automatically (essentially iterate all ResourceObjectType and set the "lowercaseclassname" to the Class) @ravening ?

Reply:
@rohityadavcloud optimized it. review it again.